eisbuk-admin/src/components/atoms/CustomerList/CustomerGrid.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+
+import TextField from "@material-ui/core/TextField";
+
+import makeStyles from "@material-ui/core/styles/makeStyles";
+
+import { Customer } from "eisbuk-shared";
+
+import CustomerGridItem from "./CustomerGridItem";
+
+import CustomerDialog from "./CustomerDialog";
+
+interface Props {
+  customers?: Customer[];
+  className?: string;
+  tableContainerClassName?: string;
+}
+
+const CustomerGrid: React.FC<Props> = ({ customers, className = "" }) => {
+  const classes = useStyles();
+
+  // search flow
+  const [searchString, setSearchString] = useState("");
+  const searchRegex = new RegExp(searchString, "i");
+
+  const [dialogCustomer, setDialogCustomer] = useState<Customer>(
+    {} as Customer
+  );
+
+  const [open, setOpen] = React.useState(false);
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const onCustomerClick = (customer: Customer) => {
+    setOpen(true);
+    setDialogCustomer(customer);
+  };
+
+  return (
+    <div className={className}>
+      <SearchField {...{ searchString, setSearchString }} />
+
+      <div className={classes.container}>
+        {customers?.map(
+          (customer, i) =>
+            (searchRegex.test(customer.name) ||
+              searchRegex.test(customer.surname)) &&
+            !customer.deleted && (
+              <CustomerGridItem
+                key={customer.id || `temp-key-${i}`}
+                onClick={onCustomerClick}
+                {...customer}
+              />
+            )
+        )}
+      </div>
+      <CustomerDialog
+        open={open}
+        onClose={handleClose}
+        customer={dialogCustomer}
+      ></CustomerDialog>
```

Comment:
`<CustomerDialog {props} />` is fine (as opposed to `<CustomerDialog></CustomerDialog>`)